pkg/asset/machines/aws/instance_types.go

```diff
@@ -0,0 +1,54 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	awsconfig "github.com/openshift/installer/pkg/asset/installconfig/aws"
+)
+
+// PreferedInstanceType returns a preferred instance type from the list of instance types provided in descinding order of preference
+// based on filters like the list of required availability zones.
+func PreferedInstanceType(ctx context.Context, meta *awsconfig.Metadata, types []string, zones []string) (string, error) {
```

Comment:
nit: `Prefered` -> `Preferred`.  Or `DefaultInstanceType`?

Reply:
fixed to Preferred https://github.com/openshift/installer/pull/3051/commits/ad333dd80b55f246edb62f2d694a194bdbc7c660